### PR TITLE
convert crdt tests to java tests

### PR DESCRIPTION
**Pull request by mmandlis**
previously were called from WebHarnessController and removed as part of https://github.com/PolymerLabs/arcs/pull/3645

### CODE REVIEW

File: src/javaharness/javatests/arcs/android/impl/CrdtTest.java [hunk 1]
```diff
@@ -140,62 +135,11 @@ public static void testRejectRemoveTooOld() {
...
-        : "modelChange2.changeType should be ChangeType.MODEL";
-    assert collectionDeepEquals(
-            (CollectionData<Data>) modelChange2.modelPostChange.get(), expectedSet2)
-        : "Unexpected merge model change (2).";
-    assert changeDeepEquals(modelChange2, (CollectionChange<Data>) result2.otherChange)
-        : "modelChange2 must be equal otherChange2";
+  @Test
+  public void testMergeModels() {
+    // This test cannot be moved, because it heavily uses package private methods and data members.
+    // TODO: refactor?
```
Comment (csilvestrini) [inline src/javaharness/javatests/arcs/android/impl/CrdtTest.java]: These tests should live in the same package as the class(es) they're testing (just swapping out java/ for javatests/). It's ok to test package-private methods/members.

This file should probably live in javatests/arcs/crdt/

File: src/javaharness/java/arcs/crdt/CrdtTestHelper.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      return this.id.equals(((Data) other).id);
+    }
+
+    @Override
+    public int hashCode() {
+      return this.id.hashCode();
+    }
+  }
+
+  public static void testMergeModels() {
```
Comment (csilvestrini) [inline src/javaharness/java/arcs/crdt/CrdtTestHelper.java]: See other comment. This file probably shouldn't exist (or possibly it should, but the test methods should belong in the test file).

If you can't put the CrdtTestHelper.Data class in the same test file, you can leave this as a test helper class, but it should ideally live inside a .testing subpackage

File: src/javaharness/javatests/arcs/android/impl/CrdtTest.java (hunk 1 above)
Comment (mmandlis) [reply src/javaharness/javatests/arcs/android/impl/CrdtTest.java]: moved to the right package

File: src/javaharness/java/arcs/crdt/CrdtTestHelper.java (hunk 2 above)
Comment (mmandlis) [reply src/javaharness/java/arcs/crdt/CrdtTestHelper.java]: deleted this file

**Comment by mmandlis** [comment]
address comments, PTAL

**Pull request merged**